src/main/java/com/mongodb/hibernate/jdbc/MongoParameterRecognizer.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.jdbc;
+
+class MongoParameterRecognizer {
+
+    static String replace(String json) {
+        StringBuilder builder = new StringBuilder(json.length());
```

Comment:
unless `json` doesn't contain `?`, this `StringBuilder` will grow for the replacement length is longer than `?`. How to assign the optimal size is unclear here

Reply:
Yeah, not sure it matters too much though